a8c-files.php

```diff
@@ -542,6 +545,18 @@ function get_files_service_hostname() {
 		return 'https://' . FILE_SERVICE_ENDPOINT;
 	}
 
+	public function upload_url_path( $upload_url_path ) {
+		// No modifications needed outside multisite.
+		if ( false === is_multisite() ) {
+			return $upload_url_path;
+		}
+		// Change the upload url path to site's URL + wp-content/uploads without trailing slash
+		// Related core code: https://core.trac.wordpress.org/browser/tags/4.6.1/src/wp-includes/functions.php#L1929
+		$upload_url_path = trailingslashit( get_site_url() ) . 'wp-content/uploads';
```

Comment:
An easier approach is to pass `wp-content/uploads` as the second argument to `get_site_url()`, leaving the first argument `null` to let Core sort out the blog ID. This likely eliminates the need for `trailingslashit()`, though replacing it with `untrailingslashit()` may be necessary/advantageous.